### ISSUE TITLE
package.json: check dependencies in deep

### ISSUE THREAD

**Issue by bugsounet** [KristjanESPERANTO/MagicMirror-3rd-Party-Modules#39]
Hi Kristjan,

MM v2.30.0 will allows to install and use modules in sub/sub folders

You can make some tests with `MMM-GoogleAssistant` dev branch
EXTs will be integrated into MMM-GoogleAssistant in the [EXTs folder](https://github.com/bugsounet/MMM-GoogleAssistant/tree/dev/EXTs) with a package.json file

**Comment by KristjanESPERANTO** [comment]
Interesting. In the long run, will you close the repositories for the different modules?

**Comment by bugsounet** [comment]
Yes, i will close old EXTs modules repos when i will release v9.x

I'm tired of updating the same dependencies on each modules ;)
with common dependencies it's less work!

This is one of the reasons why I will use this new MM patch

This will allow you to prepare on your side

Happy coding,
Cédric

**Comment by KristjanESPERANTO** [comment]
Now your modules are no longer listed individually in "my" module list. That's what you mainly want back, right? We can certainly implement that.

Apart from that: I think it's really nice that you've integrated the EXT modules directly into MMM-GoogleAssistant, it makes it much easier for users to use the modules. Wouldn't it be possible to omit the installation part for each EXT module if you put the dependencies directly into MMM-GoogleAssistant? Then users would only have to extend the config to use a module.

**Comment by bugsounet** [comment]
>Now your modules are no longer listed individually in "my" module list. That's what you mainly want back, right? We can certainly implement that.

Yes, all in one with `npm run` commands


> Wouldn't it be possible to omit the installation part for each EXT module if you put the dependencies directly into MMM-GoogleAssistant?

humm, it has many modules that require sub-dependencies via apt to be installed and then use electron-rebuild...
I'm afraid that an error will occur and prevent the installation of MMM-GoogleAssistant
I had made this system in the past with v3 or v4. I had too many issues

> Then users would only have to extend the config to use a module.

It's automated a 90% with `EXT-Website`
Maybe it can be again more simple with [moduleConfig](https://github.com/MagicMirrorOrg/MagicMirror/pull/3682) feature (and another module positioning order feature that will never be released as well...)

Sorry for my (very very very) bad english

**Comment by bugsounet** [comment]
Hi Kristjan,

Sorry, this issue is no longer valid.
I just removed all my modules from the "3rd party modules"
So I decided not to code for MM² anymore

it's not your fault, don't worry

Sorry again for my English (not understandable).
I will close it

**Comment by KristjanESPERANTO** [comment]
That's a pity, you have made many valuable contributions. Thank you for that. Working in an international team with very different characters is not always easy - I understand that.  And I don't think your English is bad at all.

I wish you all the best with the new projects you choose next. There is so much to discover and learn.
